File: data/transcoder_evaluation_gfg/go/WRITE_AN_EFFICIENT_METHOD_TO_CHECK_IF_A_NUMBER_IS_MULTIPLE_OF_3.go
package main

import (
	"math"
		"fmt"
	"os"
	"unsafe"
)

func min(x int, y int) int {
	if x < y {
		return x
	}
	return y
}
func max(x int, y int) int {
	if x > y {
		return x
	}
	return y
}
func cmpfunc(a unsafe.Pointer, b unsafe.Pointer) int {
	return *(*int)(a) - *(*int)(b)
}


func f_gold(n int) int {
	var (
		odd_count  int = 0
		even_count int = 0
	)
	if n < 0 {
		n = -n
	}
	if n == 0 {
		return 1
	}
	if n == 1 {
		return 0
	}
	for n != 0 {
		if n&1 != 0 {
			odd_count++
		}
		if n&2 != 0 {
			even_count++
		}
		n = n >> 2
	}
	return f_gold(int(math.Abs(float64(odd_count - even_count))))
}
//TOFILL
func main() {
	var (
		n_success int     = 0
		param0    []int = []int{94, 94, 79, 39, 16, 90, 64, 76, 83, 47}
	)
	for i := int(0); i < len(param0[:]); i++ {
		if f_filled(param0[i]) == f_gold(param0[i]) {
			n_success += 1
		}
	}
	fmt.Print("#Results:", " ", n_success, ", ", len(param0[:]))
	os.Exit(0)
}

File: data/transcoder_evaluation_gfg/rust/WRITE_AN_EFFICIENT_METHOD_TO_CHECK_IF_A_NUMBER_IS_MULTIPLE_OF_3.rs
#![feature(main)]
// Copyright (c) 2019-present, Facebook, Inc.
// All rights reserved.
//
// This source code is licensed under the license found in the
// LICENSE file in the root directory of this source tree.
//

#[no_mangle]
pub unsafe fn abs(mut x: i32)
 -> i32 {
    return x.abs();
}

#[no_mangle]
pub unsafe fn min(mut x: i32, mut y: i32)
 -> i32 {
    return if x < y { x } else { y };
}
#[no_mangle]
pub unsafe fn max(mut x: i32, mut y: i32)
 -> i32 {
    return if x > y { x } else { y };
}

pub unsafe fn f_gold(mut n: i32) -> i32 {
    let mut odd_count: i32 = 0 as i32;
    let mut even_count: i32 = 0 as i32;
    if n < 0 as i32 { n = -n }
    if n == 0 as i32 { return 1 as i32 }
    if n == 1 as i32 { return 0 as i32 }
    while n != 0 {
        if n & 1 as i32 != 0 { odd_count += 1 }
        if n & 2 as i32 != 0 { even_count += 1 }
        n = n >> 2 as i32
    }
    return f_gold(abs(odd_count - even_count));
}
//TOFILL
unsafe fn main_0() -> i32 {
    let mut n_success: i32 = 0 as i32;
    let mut param0: [i32; 10] =
        [94 as i32, 94 as i32, 79 as i32,
         39 as i32, 16 as i32, 90 as i32,
         64 as i32, 76 as i32, 83 as i32,
         47 as i32];
    let mut i: i32 = 0 as i32;
    while i < param0.len() as i32 {
        if f_filled(param0[i as usize]) == f_gold(param0[i as usize]) {
            n_success += 1 as i32
        }
        i += 1
    }
    println!("{} {} {} {} {}", "#Results:", " ", n_success, ", ", param0.len() as i32);
    return 0 as i32;
}
#[main]
pub fn main() { unsafe { ::std::process::exit(main_0() as i32) } }
